src/tdastro/astro_utils/passbands.py

```diff
@@ -56,44 +77,92 @@ def __str__(self) -> str:
+
+    def process_transmission_tables(
+        self, delta_wave: Optional[float] = 5.0, trim_quantile: Optional[float] = 1e-3
+    ):
+        """Process the transmission tables for all passbands in the group; recalculate group's wave attribute.
+
+        Parameters
+        ----------
+        delta_wave : float or None, optional
+            The grid step of the wave grid. Default is 5.0.
+        trim_quantile : float or None, optional
+            The quantile to trim the transmission table by. For example, if trim_quantile is 1e-3, the
+            transmission table will be trimmed to include only the central 99.8% of rows.
+        """
+        for passband in self.passbands.values():
+            passband.process_transmission_table(delta_wave, trim_quantile)
+
+        self._update_waves()
+
+    def fluxes_to_bandfluxes(self, flux_density_matrix: np.ndarray) -> np.ndarray:
         """Calculate bandfluxes for all passbands in the group.
 
         Parameters
         ----------
         flux_density_matrix : np.ndarray
             A 2D array of flux densities where rows are times and columns are wavelengths.
-        wavelengths_angstrom : np.ndarray
-            An array of wavelengths in Angstroms corresponding to the columns of flux_density_matrix.
 
         Returns
         -------
-        dict
-            A dictionary of bandfluxes with passband labels as keys.
+        dict of np.ndarray
+            A dictionary of bandfluxes with passband names as keys and np.ndarrays of bandfluxes as values.
         """
+        if flux_density_matrix.size == 0 or len(self.waves) != len(flux_density_matrix[0]):
+            flux_density_matrix_num_cols = 0 if flux_density_matrix.size == 0 else len(flux_density_matrix[0])
+            raise ValueError(
+                f"PassbandGroup mismatched grids: Flux density matrix has {flux_density_matrix_num_cols} "
+                f"columns, which does not match transmission table's {len(self.waves)} rows. Check that the "
+                f"flux density matrix was calculated on the same grid as the transmission tables, which can "
+                f"be accessed via the Passband's or PassbandGroup's waves attribute."
+            )
+
         bandfluxes = {}
-        for label, passband in self.passbands.items():
-            bandfluxes[label] = passband.fluxes_to_bandflux(flux_density_matrix, wavelengths_angstrom)
+        for full_name, passband in self.passbands.items():
+            # We only want the fluxes that are in the passband's wavelength range
+            # So, find the indices in the group's wave grid that are in the passband's wave grid
+            lower, upper = passband.waves[0], passband.waves[-1]
+            lower_index, upper_index = np.searchsorted(self.waves, [lower, upper])
+            # Check that this is the right grid after all (check will fail if the grid is not regular, or
+            # passbands are overlapping)
```

Comment:
I'm a little confused why the check would fail in these scenarios.

Reply:
**Units**: I do agree--but for now I did preserve the code Micheal wrote, feeling that refactoring it to be out of scope of the goal of this PR.  I've recorded details for unit changes we want to make in this issue: #117

**Check:** In most cases, we shouldn't need this; however, if the user supplies transmission tables that overlap and do not share the same step (and the user specifies delta_wave=None), the values from each would be interleaved. 

eg, (and going into this level of detail just to kind of think out loud/review this for myself here)
imagine a group with two passbands:
`passband_a.waves = [2, 4, 6]`
`passband_b.waves = [5, 6, 7, 8]` 
which results in:
`group.waves = [2, 4, 5, 6, 7, 8]`
and if we just grabbed waves based on lower and upper bounds, we would get end up grabbing that extra "5" value when generating our indices:
`[2, 4, 5, 6, 7, 8]`
`[1, 1, 1, 1, 0, 0]` <-- mask representing the indices we get
when what we really want is:
`[2, 4, 5, 6, 7, 8]`
`[1, 1, 0, 1, 0, 0]` <--excluding the 5
It's a small edge case, but possible to encounter under the specs I've been given (my understanding from the last time I spoke to @hombit Kostya is that we would like to support non-uniform transmission tables, but I asked him a while ago and fairly informally, so it could be worth revisiting). 
The check is at worse O(n), but if we can guarantee uniform table steps we could get rid of this entirely.

Reply:
Thanks! That makes sense to me now. I think my confusion is does the check fails on one of the conditions or it only fails on both conditions (non-uniform grid AND overlapping). My understanding is the latter.

Reply:
(but the doc implies the former?)